torch/csrc/utils/python_arg_parser.h

```diff
@@ -250,7 +254,17 @@ inline std::vector<int64_t> PythonArgs::intlistWithDefault(int i, std::vector<in
   for (int idx = 0; idx < size; idx++) {
     PyObject* obj = tuple ? PyTuple_GET_ITEM(arg, idx) : PyList_GET_ITEM(arg, idx);
     try {
-      res[idx] = THPUtils_unpackIndex(obj);
+      // Elements of torch.Size are tensors during tracing, and we need to record extra
+      // information before they are turned into an IntList
+      if (traceable && reinterpret_cast<PyObject*>(Py_TYPE(obj)) == THPVariableClass) {
```

Comment:
why don't you use THPVariable_Check and THPVariable_Unpack here?

Reply:
I used that, because this case is extremely rare, and I don't want to have `PyObject_IsInstance` in the hot path of the arg parser, so I used a cheaper pointer comparison. I can change that to `THPVariable_Check` if you really want me to.

Reply:
Well, we have THPVariable_Check all over the place in the arg parser; I'm okay either way as long as we do the same thing consistently.

Reply:
The difference is that other places where we do it are tensors or scalar arguments, not IntLists where we do it for every item. wdyt?